sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -423,10 +434,18 @@ def refresh(self, **kwargs) -> None:
                             key,
                             label,
                         )
-                        self._load_all(**kwargs)
-                        self._refresh_on[(key, label)] = updated_sentinel.etag
-                        self._refresh_timer.reset()
-                        return
+                        need_refresh = True
+
+                        updated_sentinel_keys[(key, label)] = updated_sentinel.etag
+                # Need to only update once, no matter how many sentinels are updated
+                if need_refresh:
+                    self._load_all(**kwargs)
```

Comment:
Same here as my other comment. `_load_all` will only add the selected key-values to `self._dict`. Sentinel keys also need to be processed and added to application configuration.